MQTT-Support

Ranlvor opened the pull request:
Ohai,

wie wir alle wissen ist polling doof und ich wollte schon immer die Push-State-API. Jetzt habe ich MQTT gefunden und damit ist genau das schön umsetzbar. Bitte mergen, in der global.php ein Passwort setzen und mir zukommen lassen.

Sobald das hier gemerged ist stelle ich die XMPP- und IRC-Bots auf MQTT um für weniger Polling und weniger Latenz.

Aktuell akzeptiert der Server jede Username/Passwort-Kombination und gibt jedem alle Rechte. Ich möchte allerdings trotzdem alle Clients mit Usernamen und Passwörtern versehen, falls wir das irgendwann mal brauchen. Ich hab auch vor eine Websocket-Schnittstelle einzurichten damit man auch in Webanwendungen den Push-Status nutzen kann.

Ranlvor committed:
update.php: added MQTT-Support

Ranlvor committed:
refactor MQTT-settings to global.php

Ranlvor committed:
set MQTT-Clientid

Because something does not act in respect of the specification. The specification allows an empty clientid, but in our setup the empty clientid causes problems.

problame commented:
Es wäre vielleicht schöner, die Daten, die MD-spezifischen Daten, die gerade in `global.php` stehen, in ein config-file auszulagern.

Oder wir lassen das mit der (theoretischen) Wiederverwendbarkeit bleiben... (in diesem Fall kann dann auch das `data/spaceapi_template.json.sample` in `data/spaceapi_template.json` umbenannt werden...

Ranlvor commented:
Ich dachte die global.php wäre das Configfile. *refactor*

Ranlvor committed:
refactor MQTT to use external configfile

For better reusability refactor MQTT-Settings to config.php and document that you perhaps want to adjust them

Ranlvor commented:
MQTT ist jetzt optional, default aus, hat ein Configfile und das Configfile ist in der README erwähnt und ein wenig kommentiert.

problame commented:
Magst du `phpMQTT` vielleicht noch auf das nötigste zusammenstutzen? oder erlaubt das die lizenz nicht?

Ranlvor commented:
Die Lizenz würde es erlauben, aber wozu?
